agent/src/capability/configstore.cpp

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (C) 2016 Pelagicore AB
+ *
+ * Permission to use, copy, modify, and/or distribute this software for
+ * any purpose with or without fee is hereby granted, provided that the
+ * above copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+ * WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+ * BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+ * OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+ * WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ * ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+ * SOFTWARE.
+ *
+ * For further information see LICENSE
+ */
+
+#include "configstore.h"
+
+
+ConfigStore::ConfigStore()
+{
+}
+
+ConfigStore::ConfigStore(const std::string &inputPath)
+{
+    ReturnCode retval = ReturnCode::SUCCESS;
+
+    if (inputPath.empty()) {
+        //File path is empty, this is ok
+        return;
+    } else if (isDirectory(inputPath)) {
```

Comment:
In the cases were we return in an if statement, I think it is nice to not use an else clause. In cases such as this, consider using another if statement. This makes it separated and more clear that the function (or constructor in this case) terminates.